#BN-9 feat: Two factor authentication for vendors

EvelyneUmubyeyi opened the pull request:
# PR Description
In this PR, I created endpoints for sending an email to the user who logged in using email and password. The sent email contains TOTP (time-based one-time password) that lasts a few minutes. One endpoint sends the email, and the other endpoint is for verifying if the TOTP entered by user for two factor authentication is valid. This PR also includes tests and documentation for those endpoints.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] 2FA option is enabled on vendor login
- [x] Appropriate error messages are returned on unsuccessful attempts
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. Test the endpoints in [ swagger ] ( http://localhost:9090/api-docs/ )
4. run unit tests in terminal using ` npm run test `
# Screenshots (If appropriate/applicable)
![image](https://user-images.githubusercontent.com/88635088/232448219-17d2c196-3e2c-4b27-945c-3becd69ce9cf.png)
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

UmubyeyiEvelyne committed:
chore(#BN-9: Add two factor authenrtication for vendors): Added create user and created a secret token that will be used to generate an OTP sent to the user via SMS

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): I added sending OTP message to the user upon successful registration using twilio

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): I added endpoints for sending OTP message to user's email, and another one for validating OTP submitted by the user
- I added documentation for these two endpoints

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): changed the start script in package.json

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): added documentation for the verify OTP endpoint

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): added tests for sendOtp endpoint

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): added tests for validate OTP endpoint

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): changed the dev script in package.json

Honore123 commented:
Resolve conflicts!

Honore123 commented on src/database/models/orderItems.js:
Use Prettier eslint extension in VSCode

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): added french translations for the messages returned in API responses

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): resolved conflicts with the develop branch

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): resolved conflicts with the develop branch

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): resolved eslint conflicts in orderItems.js

uwdjany reviewed (dismissed):
Well done!

Honore123 commented on src/middleware/verifyJWT.js:
Translate response message in this file

Honore123 commented on src/utils/emailValidator.js:
It will be better if you delete those comments

UmubyeyiEvelyne committed:
feat(#BN-9: Two factor authentication for vendors): worked on requested changes:
removed unnecessary comments in emailValidator.js
translated API response messages in verifyJWT.js
